ocaml/xapi/xapi_sr.ml

```diff
@@ -387,15 +387,15 @@ let update_vdis ~__context ~sr db_vdis vdi_infos =
       | loc, None, Some v -> Some v
       | _, _, _ -> None
     ) db_vdi_map scan_vdi_map in
-  let to_update = StringMap.merge (fun loc db scan -> match loc, db, scan with
-      | loc, Some (r, v), Some vi -> Some (r, v, vi)
-      | _, _, _ -> None
-    ) db_vdi_map scan_vdi_map in
 
   let find_vdi db_vdi_map loc =
     if StringMap.mem loc db_vdi_map
     then fst (StringMap.find loc db_vdi_map)
-    else Ref.null in
+    else
+      (* CA-254515: Also check for the snapshoted VDI in the database *)
+      try Db.VDI.get_by_uuid ~__context ~uuid:loc
```

Comment:
So probably everything would work if we leave this as it is, but right now it's a bit confusing, because we call the parameter `loc`, but we use it as a UUID here, but it's not always the same as a UUID. (So strictly speaking we should use `get_..._where` to look up the VDI based on its location.)
However, to make it even more confusing, it seems that the `snapshot_of` field is a location according to xcp-idl: https://github.com/xapi-project/xcp-idl/blob/master/storage/storage_interface.ml#L64 , but xapi actually sets it to  UUID: https://github.com/xapi-project/xen-api/blob/master/ocaml/xapi/storage_access.ml#L100
So I'm really confused now, so maybe let's leave it as it is :S